tests/test_fixtures.py

```diff
@@ -12,3 +12,11 @@ def test_s3_versioning(tmp_s3_path, versioning):
 
 def test_s3_versioning_disabled(tmp_s3_path):
     assert not tmp_s3_path.fs.version_aware
+
+
+def test_s3_server_default_config(s3_server):
+    assert "endpoint_url" in s3_server
+    assert s3_server["aws_access_key_id"] == "pytest-servers"
```

Comment:
Q: how about profile, and security token? do we need to tests for them?

Reply:
do we need to include the into the "client_kwargs" dict ?

Reply:
The main reason why I ended up going for this `client_kwargs` dict is that by looking at the boto3 [credentials docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html), it mentions that the order it uses to search for credentials is the following:

1. **Passing credentials as parameters in the boto.client() method**
2. Passing credentials as parameters when creating a Session object
3. Environment variables
4. Shared credential file (~/.aws/credentials)
5. AWS config file (~/.aws/config)
6. Assume Role provider
7. Boto2 config file (/etc/boto.cfg and ~/.boto)
8. Instance metadata service on an Amazon EC2 instance that has an IAM role configured.

This means that if we pass `aws_session_token` `aws_secret_access_key` and `aws_access_key_id`, it should be enough:

- security token is an environment variable and has lower priority. From the docs 
```The AWS_SECURITY_TOKEN environment variable can also be used, but is only supported for backwards compatibility purposes. ```
- Profile is used to parse session token, secret access key and id from the .aws credentials/config files, which also have lower priority


Regarding the `client_kwargs` dict, we could make it more explicit and pass `key`, `secret`, `token` etc as kwargs, since UPath passes these directly as `S3FileSystem` init kwargs (https://s3fs.readthedocs.io/en/latest/api.html#s3fs.core.S3FileSystem), but I was a bit concerned in introducing yet another adapter, plus with client_kwargs we could eventually configure the underlying client with options that S3FileSystem hasn't exposed yet

Reply:
Good. Make sense.